tests/test_parsing.py

```diff
@@ -1,21 +1,58 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-"""
+'''
 test_iniparse
 ----------------------------------
 
 Tests for `inicheck.iniparse` module.
-"""
-
-import unittest
-
+'''
+import pytest
+from collections import OrderedDict
 from inicheck.iniparse import *
 
 
-class TestIniparse(unittest.TestCase):
-
-    def test_parse_sections(self):
-        """
+class TestIniparse():
+
+    def run_parsing_test(self, fn, info, expected, exception):
+        '''
+        Manages testing exceptions and testing dict output
+        '''
+        if type(info) == OrderedDict:
+            expected = OrderedDict(expected)
+
+        if exception == None:
+            received = fn(info)
+
+            # Test the dictionaries recieved match the expected
+            assert received == expected
+
+        else:
+            with pytest.raises(exception):
+                received = fn(info)
+
+    @pytest.mark.parametrize('info, expected, exception', [
+    # Test a simple parsing
+    (['[s]','i:v'], {'s':['i:v']}, None),
+    # Test a section only without items
+    (['[s]'], {'s':[]}, None),
+    # Test every section is made lower case
+    (['[CASE]'], {'case':[]}, None),
+    # Test space remove before and after section
+    (['[ spaces ]'], {'spaces':[]}, None),
+    # Test a single line section/value is parsed
+    (['[single_line]i:v'], {'single_line':['i:v']}, None),
+    # Test some comment removal
+    (['#comment','[s]','i:v'], {'s':['i:v']}, None),
+    ([';[commented]'], {}, None),
+    (['[s]','i:v ;comment'], {'s':['i:v']}, None),
+    # Test exception with repeat sections
+    (['[test]', '#', '[test]'], {}, Exception),
+    # Test non-comment chars before the first section
+    (['a#', '#', '[test]'], {}, Exception),
+
+    ])
+    def test_parse_sections(self, info, expected, exception):
```

Comment:
This is personal preference and I would split between error and successful test cases. Makes it easier to read and identify what is handled or not. Also removes the need for a method that needs logic to case between arguments.

Reply:
That sounds good to me. You are thinking test_parse_sections and adding say a test_parse_sections_exceptions?

Reply:
Yup, something along those lines